lib/Reconciler.spec.lua

```diff
@@ -85,4 +89,62 @@ return function()
 		Reconciler.unmount(handle)
 		expect(bRef.current).to.never.be.ok()
 	end)
+
+	it("should clean up Event references properly", function()
+		local sizeChangedCount = 0
+
+		local function eventCallback(object, property)
+			if property == "Size" then
+				sizeChangedCount = sizeChangedCount + 1
+			end
+		end
+
+		local element = createElement("Frame", {
+			[Event.Changed] = eventCallback,
+		})
+
+		local handle = Reconciler.mount(element, ReplicatedStorage, "Foo")
```

Comment:
Let's mount into a container that we create in the test so that we don't leave cruft in the DataModel in the case the test fails.